Frontend/src/ui/Sidebar.jsx

```diff
@@ -1,19 +1,26 @@
 import { Sidebar } from '@thewca/wca-components'
-import React from 'react'
+import React, { useContext } from 'react'
 import { useParams } from 'react-router-dom'
+import { canAdminCompetition } from '../api/auth/get_permissions'
+import { AuthContext } from '../api/helper/context/auth_context'
 
 export default function PageSidebar() {
   const { competition_id } = useParams()
+  const { user } = useContext(AuthContext)
+  const sideBarItems = []
```

Comment:
This is poorly named, it's only 1 (or 0) of the several side bar items.

It's also not very react-like to define an array/object and then mutate it later (at least, not at the top level of the component; it might be done in a helper function whose sole purpose is to create that array/object).

Probably the most standard way around this would be for `Sidebar` to accept `SidebarItem`s as children:
```
return (
  <Sidebar>
    {canAdmin... && <SidebarItem title={...} iconName={...} ... />} // registration
    <SidebarItem ... /> // register
    <SidebarItem ... /> // competitors
  </ Sidebar>
  )
```
But that involves `wca-components` so would probably be in a separate PR, if it's something we want to do.

Reply:
I agree. The sidebar component might not survive the redesign, but I will change it and mark it as a todo